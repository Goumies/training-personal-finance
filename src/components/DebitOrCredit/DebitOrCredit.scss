.debit-or-credit {
  display: inline-grid;
  width: 200px;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: 1em;

  .credit {
    grid-column: 1 / 2;
    text-align: right;
  }

  .debit {
    grid-column: 2 / 3;
    text-align: right;
  }
}
