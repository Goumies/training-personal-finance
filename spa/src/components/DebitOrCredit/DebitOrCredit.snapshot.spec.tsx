import React from "react";
import renderer from "react-test-renderer";
import { DebitOrCredit, Props } from "./DebitOrCredit";

it("should render positive amounts on the left side", () => {
  const props: Props = {
    amount: 10
  };
  const tree = renderer.create(<DebitOrCredit {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render zero on the left side", () => {
  const props: Props = {
    amount: 0
  };
  const tree = renderer.create(<DebitOrCredit {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render negative amounts on the right side", () => {
  const props: Props = {
    amount: -10
  };
  const tree = renderer.create(<DebitOrCredit {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
