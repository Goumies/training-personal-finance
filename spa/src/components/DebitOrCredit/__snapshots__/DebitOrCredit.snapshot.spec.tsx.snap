// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render negative amounts on the right side 1`] = `
<div
  className="debit-or-credit"
>
  <span
    className="credit"
  />
  <span
    className="debit"
  >
    <span
      className="value"
    >
      -10
    </span>
  </span>
</div>
`;

exports[`should render positive amounts on the left side 1`] = `
<div
  className="debit-or-credit"
>
  <span
    className="credit"
  >
    <span
      className="value"
    >
      +10
    </span>
  </span>
  <span
    className="debit"
  />
</div>
`;

exports[`should render zero on the left side 1`] = `
<div
  className="debit-or-credit"
>
  <span
    className="credit"
  >
    <span
      className="value"
    >
      0
    </span>
  </span>
  <span
    className="debit"
  />
</div>
`;
